File: src/main/java/historywowa/global/infra/scheduled/ScheduledService.java
package historywowa.global.infra.scheduled;

import historywowa.domain.point.domain.repository.PointRepository;
import jakarta.transaction.Transactional;
import lombok.RequiredArgsConstructor;
import org.hibernate.annotations.Comment;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;

@Component
@Transactional
@RequiredArgsConstructor
public class ScheduledService {

    private final PointRepository pointRepository;

    @Scheduled(cron = "0 0 0 * * *", zone = "Asia/Seoul")
    public void resetPointsAtMidnight() {
        pointRepository.updateAllBalanceToFixedValue(2);
    }

}

File: src/main/kotlin/historywowa/global/infra/scheduled/ScheduledService.kt
package historywowa.global.infra.scheduled

import historywowa.domain.point.domain.repository.PointRepository
import jakarta.transaction.Transactional
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
@Transactional
class ScheduledService(
        private val pointRepository: PointRepository
) {

    @Scheduled(cron = "0 0 0 * * *", zone = "Asia/Seoul")
    fun resetPointsAtMidnight() {
        pointRepository.updateAllBalanceToFixedValue(2)
    }
}
